### ISSUE TITLE
title-style boxed bad behaviour when title is too long

### ISSUE THREAD

**Issue by enseignantePC** [Pablo-Gonzalez-Calderon/showybox-package#12]
minimal error example : https://typst.app/project/rHTdT0zZHdHzbsvwTOmUpS 

screenshot of the problem 
![image](https://github.com/Pablo-Gonzalez-Calderon/showybox-package/assets/82136278/c8393995-e13b-4566-b588-e576362a1986)

thanks for your pretty and useful ~crate~ package! (my usual pseudo is astrale-sharp but I'm working right now)

**Comment by Pablo-Gonzalez-Calderon** [comment]
Hi!

Hmm... It seems I've forgot some "limit" cases while making boxed-titles. Thank you for reporting this bug, I'll work on a fix in the following days/weeks (currently i'm quite running out of time for other projects I have 😅)

**Comment by enseignantePC** [comment]
If you give me some pointers I might make something !

**Comment by Pablo-Gonzalez-Calderon** [comment]
Sure!

Looking at the code fastly, I see the following changes must be considered:
1. The maximum width of the title must be controled (now it isn't). For that, an outset or inset property must be implemented for the boxed-title. **The space at each side (left and right) must be of ``1em``** to keep coherency.
2. The bad behaviour of the title only occurs when a single-line-title is too long to fit in a single line. If it's broke in parts using a ``\n`` or a ``\``, the behaviour is normal.
3. This bug won't happen in a determined type of boxed-title showybox. **It depends of the container width and the title length,** so this solution must be flexible to handle various cases for the same showybox. Therefore, the variable ``title-block-height`` must be modified properly --and probably is the cause of the bug. A declaration of the variable is shown below:
```javascript
let title-size = measure(title, styles)
let title-block-height = title-size.height + showy-inset(top, showy-section-inset("title", frame)) + showy-inset(bottom, showy-section-inset("title", frame))
```
This variable can be the cause of the bug because it calculates the title height before it's drawn and put in the showybox. There, the title isn't broken yet, because it hasn't received a max-width value. Therefore, the evaluation of the size of the title must be re-calculated in a smart way, because of it depends also the correct spacing of several parts of a boxed-title showybox.

That's all I have thought and seen at the moment. Hope you can find a solution earlier than me 😅

**Comment by Pablo-Gonzalez-Calderon** [comment]
Hi!

I'm happy to announce this bug has been successfully fixed! 🥳🥳🥳

It looked ~horrific~ complex at the beginning, but now the title rendering calculates correctly the title dimensions. Here is a image of a long-title showybox:
![image](https://github.com/Pablo-Gonzalez-Calderon/showybox-package/assets/30756770/f27c12f5-13e6-4f52-92bc-97b3768bb2f9)

The new changes will be published in the version 1.2.0 (or so) soon. I'll make a comment here to notice you when it's public and tell me if it works for you, and, finally, close this issue 🥳🥳🥳

**Comment by Pablo-Gonzalez-Calderon** [comment]
Now the changes are available here at the showybox's repo. So, I'm going to close this as it is solved. Hopefully, before October is here, the version will be up in Typst packages repo 🧐
